File: typescript/behavioral/state/guru_example/concreteStateB.ts
import {State} from "./state";
import {ConcreteStateA} from "./concreteStateA";

export class ConcreteStateB extends State {
  public handle1(): void {
    console.log('ConcreteStateB handle1, transition to ConcreteStateA')
    this.context.transitionTo(new ConcreteStateA())
  }

  public handle2(): void {
    console.log('ConcreteStateB handle2.')
  }
}

File: dart/behavioral/state/guru_example/bin/concrete_state_b.dart
import 'concrete_state_a.dart';
import 'state.dart';

class ConcreteStateB extends State {
  @override
  void handle1() {
    print('ConcreteStateB handle1.');
  }

  @override
  void handle2() {
    print('ConcreteStateB handle2, transitionTo ConcreteStateA.');
    context.transitionTo(ConcreteStateA());
  }
}
